fix #1832: support regex in FROM clause

dgnorton opened the pull request:
Changes:

- Change scanner / parser to accept regex in the `FROM` clause.  This part of the change is a little messy because the scanner can't recognize regex tokens on its own.  The parser has to ask the scanner to peek ahead to see if there is a regex and, if there is, tell the scanner to scan it.

- Change the AST and parser to accept multiple sources in the FROM clause.

- Remove old join / merge code from the AST & parser.

- Change the server to expand regexes in the FROM clause to individual measurements that match.

- Change the server so it doesn't produce duplicate field names when expanding wildcards now that the FROM clause accepts multiple sources from which to draw field names from.

- Change the engine so that it creates map-reduce jobs for multiple sources instead of just one.

- Fix bug with trailing semicolon on single statements.

dgnorton commented:
@pauldix @benbjohnson

otoolep commented on cmd/influxd/server_integration_test.go:
```diff
@@ -837,7 +850,7 @@ func TestSingleServer(t *testing.T) {
 	nodes := createCombinedNodeCluster(t, testName, dir, 1, 8090, nil)
 
-	runTestsData(t, testName, nodes, "mydb", "myrp")
+	runTestsData(t, testName, nodes, "mydb", "myrp", 7)
```
You don't mean to check this in, I'm sure. But I'm glad you found it useful. :-)

dgnorton replied:
Good catch!  Yes, very useful.  :)

toddboom commented:
At a high level, this looks good to me.

benbjohnson commented:
Sorry for the delayed response. Overall lgtm. The only piece I could see changing is separating out `Measurement` into two separate types: one for strings and one for regex.

dgnorton commented:
@benbjohnson I changed the implementation several times and never really liked it...finally just picked something and ran with it.  At first, I had something like you're suggesting (I think).  `Measurement` and `RegexLiteral` were separate `Source` types but a `Source` can have a `db` and `rp`.  It didn't make sense to add those to `RegexLiteral`.  I thought about creating another source type similar to `Measurement` called `MeasurementRegex` that contained the `db.rp.` string and a `RegexLiteral` but then I thought that was unnecessarily complicated.  So, I just added a `RegexLiteral` member to `Measurement`.  Is any of that inline with what you were thinking?
